cpp/include/rapids_triton/model/shared_state.hpp

```diff
@@ -119,6 +119,23 @@ struct SharedModelState {
     }
   }
 
+  auto get_output_names() const {
+    auto output_names = std::vector<std::string>{};
+    output_names.reserve(output_shapes_.size());
+    std::transform(std::begin(output_shapes_), std::end(output_shapes_), std::back_inserter(output_names), [](auto& output_shape) {
+      return output_shape.first;
+    });
+    return output_names;
+  }
+
+  auto check_output_name(std::string const& name) const -> bool {
+    auto cached_shape = std::lower_bound(
```

Comment:
I'm assuming we never figured out how to do this with `std::binary_search`? If so, would you mind just adding a `TODO` comment, and we can revisit after this PR?